0.download_data/scripts/0.download_MNIST_data.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# This noteboook download the MNIST dataset
+
+# In[1]:
+
+
+import pathlib
+import pickle
+
+# mnist data parser import
+import mnist
+
+# In[2]:
+
+
+mnist.datasets_url = "https://storage.googleapis.com/cvdf-datasets/mnist/"
+
+# make the mnist directory
+mnist_path = pathlib.Path("../../data/mnist")
+mnist_path.mkdir(parents=True, exist_ok=True)
+
+mnist.temporary_dir = lambda: str(mnist_path)
+
+# In[3]:
+
+
+train_images = mnist.train_images()
+train_labels = mnist.train_labels()
+test_images = mnist.test_images()
+test_labels = mnist.test_labels()
+
+# In[4]:
+
+
+print("Saving data into a pickle file ...")
```

Comment:
Just curious, why do you need to save as a pickle file instead of downloading all the images as TIFFs? This might be specific to the mnist dataset.

Reply:
I thought the same... the MNIST data set is smaller in the pickle file  than in images due to its structure.